From ebb4e195b94074cb66d031d1f449d55cba86a182 Mon Sep 17 00:00:00 2001
From: Rogerio Guerra Borin <rogerio.borin@toradex.com>
Date: Tue, 6 Feb 2024 23:36:15 -0300
Subject: [PATCH 2/2] mount: Allow building when macro LOOP_CONFIGURE is not
 available

This is to allow building the software on machines not having the macro
LOOP_CONFIGURE or the struct loop_config in header "linux/loop.h" (both
of which added to the kernel uapi at the same time); the code snippet
providing them was taken from package util-linux, source file
"include/loopdev.h".

Upstream-Status: Accepted [https://github.com/containers/composefs/pull/253]

Signed-off-by: Rogerio Guerra Borin <rogerio.borin@toradex.com>
---
 libcomposefs/lcfs-mount.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/composefs/libcomposefs/lcfs-mount.c b/composefs/libcomposefs/lcfs-mount.c
index 5285833..8acc7b9 100644
--- a/composefs/libcomposefs/lcfs-mount.c
+++ b/composefs/libcomposefs/lcfs-mount.c
@@ -49,6 +49,20 @@
 #include "lcfs-utils.h"
 #include "lcfs-internal.h"
 
+#ifndef LOOP_CONFIGURE
+/* Snippet from util-linux/include/loopdev.h */
+/*
+ * Since Linux v5.8-rc1 (commit 3448914e8cc550ba792d4ccc74471d1ca4293aae)
+ */
+#define LOOP_CONFIGURE 0x4C0A
+struct loop_config {
+	uint32_t fd;
+	uint32_t block_size;
+	struct loop_info64 info;
+	uint64_t __reserved[8];
+};
+#endif
+
 static int syscall_fsopen(const char *fs_name, unsigned int flags)
 {
 #if defined __NR_fsopen
-- 
2.25.1

